src/TulsiGenerator/XcodeProjectGenerator.swift

```diff
@@ -414,24 +414,28 @@ final class XcodeProjectGenerator {
                                 intermediateArtifacts: intermediateArtifacts)
   }
 
+  // Resolves the given Bazel path (which is expected to begin with external/) to a filesystem
+  // path. This is intended to be used to resolve "@external_repo" style labels to paths usable by
+  // Xcode. Returns nil if the path could not be resolved for any reason.
+  func resolveExternalReferencePath(_ path: String, xcodeProject: PBXProject) -> String {
+    // Assume that the main group is a relative path to the WORKSPACE directory
+    // When the project is generated at the WORKSPACE root, the external paths should be
+    // `tulsi-workspace/external`. In this case `mainGroup.path` returns nil
+    let relativeWorkspacePath = xcodeProject.mainGroup.path != nil ? "\(xcodeProject.mainGroup.path!)/" : ""
+    return "\(relativeWorkspacePath)\(PBXTargetGenerator.TulsiWorkspacePath)/\(path)"
+  }
+
   // Examines the given xcodeProject, patching any groups that were generated under Bazel's magical
   // "external" container to absolute filesystem references.
   private func patchExternalRepositoryReferences(_ xcodeProject: PBXProject) {
     let mainGroup = xcodeProject.mainGroup
-    guard let externalGroup = mainGroup.childGroupsByName["external"] else { return }
+    guard let externalGroup = mainGroup.childGroupsByName[".."] else { return }
     let externalChildren = externalGroup.children as! [PBXGroup]
```

Comment:
Hey @dmishe thanks for looking this over again! On my original PR, this bit got lost when I cherry picked the commit onto bazebuild/tulsi.

It looks like the key for the external group was originally stored by the key `external` in `childGroupsByName`, but now it is using `..` as the key. Without this change, the code path isn't hit. 

I'll post up a test project for this tomorrow AM which include external dependencies and has an example of this 👍